src/main/java/org/folio/edge/rtac/RtacHandler.java

```diff
@@ -1,114 +1,195 @@
 package org.folio.edge.rtac;
 
+import static io.vertx.core.http.HttpHeaders.ACCEPT;
 import static java.util.stream.Collectors.toList;
+import static org.folio.edge.core.Constants.APPLICATION_JSON;
 import static org.folio.edge.core.Constants.APPLICATION_XML;
+import static org.folio.edge.rtac.utils.RtacUtils.SEPARATOR_COMMA;
+import static org.folio.edge.rtac.utils.RtacUtils.checkSupportedAcceptHeaders;
+import static org.folio.edge.rtac.utils.RtacUtils.hasRequestAnyOfAcceptTypes;
+import static org.folio.edge.rtac.utils.RtacUtils.isXmlRequest;
+import static org.folio.edge.rtac.utils.RtacUtils.returnEmptyResponse;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
-
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.folio.edge.core.Handler;
 import org.folio.edge.core.security.SecureStore;
 import org.folio.edge.core.utils.Mappers;
 import org.folio.edge.rtac.model.Holdings;
 import org.folio.edge.rtac.model.Instances;
+import org.folio.edge.rtac.utils.RtacMimeTypeEnum;
 import org.folio.edge.rtac.utils.RtacOkapiClient;
 import org.folio.edge.rtac.utils.RtacOkapiClientFactory;
 
-import com.amazonaws.util.CollectionUtils;
-import com.amazonaws.util.StringUtils;
-import com.fasterxml.jackson.core.JsonProcessingException;
-
-import io.vertx.core.http.HttpHeaders;
-import io.vertx.ext.web.RoutingContext;
-
 public class RtacHandler extends Handler {
-
   private static final Logger logger = LogManager.getLogger(RtacHandler.class);
 
-  private static final String FALLBACK_EMPTY_RESPONSE = Mappers.XML_PROLOG + "\n<holdings/>";
-
   private static final String PARAM_FULL_PERIODICALS = "fullPeriodicals";
   private static final String PARAM_TITLE_ID = "mms_id";
   private static final String PARAM_INSTANCE_ID = "instanceId";
   private static final String PARAM_INSTANCE_IDS = "instanceIds";
 
+  private List<String> rtacMimeTypes;
+  private boolean needUpdateRtacMimeType;
+
   public RtacHandler(SecureStore secureStore, RtacOkapiClientFactory ocf) {
     super(secureStore, ocf);
   }
 
+  private void saveRequestType(HttpServerRequest request) {
+    rtacMimeTypes = getRtacMimeType(request);
```

Comment:
Is it safe for multi-user execution?

Reply:
Done.

Reply:
Can't we just store the accept data (XML or JSON response) in a final local variable instead of using these synchronized methods?

Reply:
I will simplify this part of code. Thank you, Mathew.

Reply:
I agree that we don't need the synchro block, I meant if it is safe to have a state for the class (the `rtacMimeTypes` and `needUpdateRtacMimeType` properties)?

Reply:
Yeah, I agree @bohdan-suprun. Given that the handler code is called asynchronously by Vert.x, I suspect that the list would be corrupted by concurrent requests. This is why I suggested a final local variable for storing the Accept header that would be in context of the thread that called the handler. Then again, I don't think we need to save/restore the Accept header at all. To be honest, I am a bit surprised by how much code/dependencies has been changed/introduced to handle this header.

Reply:
I fixed the code to just remember original headers in place in final variable and then just restore it for possible further chain processing of request, if any.